nanoFramework.Device.Bluetooth/BluetoothEventListener.cs

```diff
@@ -16,9 +16,11 @@ internal class BluetoothEventListener : IEventProcessor, IEventListener
     {
         // Map of Bluetooth Characteristic numbers to GattLocalCharacteristic objects.
         private static readonly ArrayList _characteristicMap = new ArrayList();
+        private readonly INativeDevice _nativeDevice;
 
-        public BluetoothEventListener()
+        public BluetoothEventListener(INativeDevice nativeDevice)
```

Comment:
We are introducing a breaking change here. @josesimoes and @AdrianSoundy are we ok with that?
Maybe we can use a pattern here like the one we're using for GpioController?
So something that would look like this:
```csharp
public BluetoothEventListener(INativeDevice nativeDevice == null)
{
  _shouldDispose = nativeDevice == null;
  _nativeDevice = nativeDevice != null ? nativeDevice : new OnChip();
}
```

Reply:
As we are introducing a breaking change with all this anyway, better follow the usual pattern and go with the above code from @Ellerbach.